hazelcast/src/main/java/com/hazelcast/nio/tcp/SocketAcceptorThread.java

```diff
@@ -51,26 +59,7 @@ public void run() {
             selector = Selector.open();
             serverSocketChannel.configureBlocking(false);
             serverSocketChannel.register(selector, SelectionKey.OP_ACCEPT);
-            while (connectionManager.isLive()) {
-                // block until new connection or interruption.
-                final int keyCount = selector.select();
-                if (Thread.currentThread().isInterrupted()) {
-                    break;
-                }
-                if (keyCount == 0) {
-                    continue;
-                }
-                final Set<SelectionKey> setSelectedKeys = selector.selectedKeys();
-                final Iterator<SelectionKey> it = setSelectedKeys.iterator();
-                while (it.hasNext()) {
-                    final SelectionKey sk = it.next();
-                    it.remove();
-                    // of course it is acceptable!
-                    if (sk.isValid() && sk.isAcceptable()) {
-                        acceptSocket();
-                    }
-                }
-            }
+            mainLoop(selector);
```

Comment:
What do you suggest? 

Personally I think it makes sense since the main accept event loop is executed there.